p2p/src/authenticated/discovery/types.rs

```diff
@@ -170,14 +252,32 @@ impl<C: PublicKey> PeerInfo<C> {
         )
     }
 
+    /// Create a new [InfoVerifier] with the provided configuration.
+    pub fn verifier(
+        me: C,
+        allow_private_ips: bool,
+        peer_gossip_max_count: usize,
+        synchrony_bound: Duration,
+        ip_namespace: Vec<u8>,
+    ) -> InfoVerifier<C> {
+        InfoVerifier {
+            me,
+            allow_private_ips,
+            peer_gossip_max_count,
+            synchrony_bound,
+            ip_namespace,
+        }
+    }
```

Comment:
Is it better to have it here:
- `let info_verifier = Info::verifier(...);` 

rather than as `new` in `InfoVerifier` itself?
- `let info_verifier = InfoVerifier::new(...);`